lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp

```diff
@@ -751,36 +847,94 @@ bool ProxyRenderDelegate::getInstancedSelectionPath(
         indexPath = _sceneDelegate->GetPathForInstanceIndex(rprimId, 0, nullptr);
         usdPath = _sceneDelegate->ConvertIndexPathToCachePath(indexPath);
     }
+
+    // The "Instances" point instances pick mode is not supported for
+    // USD_IMAGING_API_VERSION < 13 (core USD versions earlier than 20.05), so
+    // no setting of topLevelPath or topLevelInstanceIndex here.
 #endif
 
-    // If update for selection is enabled, we can query Maya selection list adjustment and USD kind
-    // once per selection update to avoid cost of executing MEL command or searching optionVar for
-    // each intersection.
+    // If update for selection is enabled, we can query the Maya selection list
+    // adjustment, USD selection kind, and USD point instances pick mode once
+    // per selection update to avoid the cost of executing MEL commands or
+    // searching optionVars for each intersection.
 #if defined(MAYA_ENABLE_UPDATE_FOR_SELECTION)
-    const TfToken& selectionKind = _selectionKind;
+    const TfToken&                   selectionKind = _selectionKind;
+    const UsdPointInstancesPickMode& pointInstancesPickMode = _pointInstancesPickMode;
 #ifndef UFE_V2_FEATURES_AVAILABLE
     const MGlobal::ListAdjustment& listAdjustment = _globalListAdjustment;
 #endif
 #else
     const TfToken selectionKind = GetSelectionKind();
+    const UsdPointInstancesPickMode pointInstancesPickMode = GetPointInstancesPickMode();
     const MGlobal::ListAdjustment listAdjustment = GetListAdjustment();
 #endif
 
-    // If selection kind is empty, the exact prim that gets picked from viewport should be selected
-    // thus no need to walk the scene hierarchy.
-    if (!selectionKind.IsEmpty()) {
-        UsdPrim prim = _proxyShapeData->UsdStage()->GetPrimAtPath(usdPath);
-        while (prim) {
-            TfToken kind;
-            if (UsdModelAPI(prim).GetKind(&kind) && KindRegistry::IsA(kind, selectionKind)) {
-                usdPath = prim.GetPath();
-                break;
+    UsdPrim prim = _proxyShapeData->UsdStage()->GetPrimAtPath(usdPath);
+    UsdPrim topLevelPrim = topLevelPath.IsEmpty()
+        ? UsdPrim()
+        : _proxyShapeData->UsdStage()->GetPrimAtPath(topLevelPath);
```

Comment:
Are you trying to make the validity check explicit? Otherwise, the GetPrimAtPath call can handle the validity check itself and return an invalid prim when the path is empty.

Reply:
If it is not going to be changed later, `topLevelPrim` should be a const object.

Reply:
Addressed in ca2d4cdee6afa7d5293dbd01d269664651836dbc.